lib/Dialect/Mgmt/IR/MgmtOps.td

```diff
@@ -86,4 +113,56 @@ def Mgmt_BootstrapOp : Mgmt_Op<"bootstrap"> {
   let assemblyFormat = "operands attr-dict `:` type($output)";
 }
 
+def Mgmt_AdjustScaleOp : Mgmt_Op<"adjust_scale"> {
+  let summary = "Adjust the scale of the input ciphertext to the specified scale (for BGV and CKKS)";
+
+  let description = [{
+    This is scheme-agonistic operation that adjust the scale of the input
+    ciphertext to the specified scale.
+
+    The scale is specified by the `scale` attribute.
+
+    At the time of `secret-insert-mgmt-<scheme>`, the concrete scale
+    is not known as the scheme parameter is not generated. After that pass
+    the scale defaults to _distinct_ negative numbers so it could not be lowered.
+    Further passes like `populate-scale-<scheme>` is responsible for materializing
+    the concrete scale when the scheme parameter is known.
+
+    Note that `scale` is a 64-bit integer and may not represent the
+    large scale like 2^80 or 2^90 in CKKS. Currently CKKS scale is
+    represented in logarithmic form, i.e. log2(scale).
+
+    When further lowered, it could be lowered to bgv.mul_plain
+    or ckks.mul_plain depending on the scheme.
+  }];
+
+  let arguments = (ins
+    AnyType:$input,
+    DefaultValuedAttr<I64Attr, "0">:$scale
+  );
+  let results = (outs AnyType:$output);
+  let assemblyFormat = "operands attr-dict `:` type($output)";
+
+  let hasCanonicalizer = 1;
+}
+
+def Mgmt_NoOp : Mgmt_Op<"no_op"> {
```

Comment:
Not a huge fan of the name here. Some alternative ideas:

- `mgmt.assume` or `mgmt.require`, if it is intended to be used to mark/override mgmt attributes mid-IR
- `mgmt.init` or `mgmt.seed`, if it is intended to set up initial mgmt attributes for static constants int he IR.